include/envoy/registry/registry.h

```diff
@@ -152,18 +173,57 @@ template <class Base> class FactoryRegistry : public Logger::Loggable<Logger::Id
     }
   }
 
+  /**
+   * Permanently disables the named factory by setting the corresponding
+   * factory pointer to null. If the factory is registered under multiple
+   * (deprecated) names, all the possible names are disabled.
+   */
+  static bool disableFactory(absl::string_view name) {
+    auto disable = [](absl::string_view name) -> bool {
```

Comment:
I'm not sure that I'm interpreting this right, but I don't think that `disableFactory` can or should be `constexpr` because it has global side-effects.

`disable` could be a private member function. I made it a local lambda because there are no other callers and this protects it from misuse.

Reply:
I meant `disable` can be, because the function pointer (or lambda) itself is const?

Reply:
or actually just do `erase` so you don't even need this lambda

Reply:
constexpr lambdas need c++17, however "When this specifier is not present, the function call operator will be constexpr anyway, if it happens to satisfy all constexpr function requirements".

I did make the local variable const though.